compiler/src/main/java/com/squareup/anvil/compiler/codegen/Parameter.kt

```diff
@@ -35,8 +36,31 @@ sealed interface Parameter {
     }
 }
 
+/**
+ * Returns a name which is unique when compared to the [Parameter.originalName] of the
+ * [superParameters] argument.
+ *
+ * This mimics Dagger's method of unique naming.  If there are three parameters named "foo", the
+ * unique parameter names will be [foo, foo2, foo3].
+ */
+internal fun String.uniqueName(
```

Comment:
Would `uniqueParameterName` a better function name?

Reply:
Definitely!  Especially considering that the receiver is just `String`, and it's top-level.

Thanks!